src/librustc_mir/borrow_check/nll/type_check/relate_tys.rs

```diff
@@ -266,7 +266,19 @@ impl<'cx, 'bccx, 'gcx, 'tcx> TypeRelating<'cx, 'bccx, 'gcx, 'tcx> {
                     locations: self.locations,
                 });
 
-            // FIXME all facts!
+            if let Some(all_facts) = borrowck_context.all_facts {
+                if let Some(from_location) = self.locations.from_location() {
```

Comment:
this would it seems be caught better by just inspecting the final outlives relations and convering them (as I mentioned earlier), then we wouldn't have to repeat this code